### PR TITLE
GridFilterButtons

### PR DESCRIPTION
**Pull request by matttdawson**
```
<GridFilterButtons<ITestRow>
  options={[
    {
      label: "All",
      defaultSelected: true,
    },
    {
      label: "< 30",
      filter: (row) => row.age < 30,
    },
  ]}
/>
```
image

### CODE REVIEW

File: src/components/gridFilter/GridFilterButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      <LuiButtonGroup>
+        {options.map((option, index) => (
+          <>
+            <LuiButton
+              key={`${index}`}
+              className={clsx(
+                `lui-button lui-button-secondary`,
+                selectedOption?.label === option.label && `lui-button-active`,
+              )}
+              style={{ whiteSpace: "nowrap" }}
```
Comment (matttdawson) [inline src/components/gridFilter/GridFilterButtons.tsx:36]: I don't like this.  What if they want to wrap.

Comment (jfee-linz) [reply src/components/gridFilter/GridFilterButtons.tsx:36]: Can nowrap be set using a default css class that can be restyled?  Or maybe an optional `labelRenderer(option, {isSelected, setSelected})` property to override the LuiButton (using radio/similar)?

File: src/components/gridFilter/GridFilterButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  const [selectedOption, setSelectedOption] = useState(options.find((option) => option.defaultSelected));
+
+  const filter = useMemo(() => selectedOption?.filter, [selectedOption]);
+  useGridFilter(filter);
+
+  return (
+    <div className="lui-margin-top-xxs lui-margin-bottom-xxs">
+      <LuiButtonGroup>
+        {options.map((option, index) => (
+          <>
```
Comment (jfee-linz) [inline src/components/gridFilter/GridFilterButtons.tsx:29]: is the fragment needed here?

File: src/components/gridFilter/GridFilterButtons.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    filter?: GridFilterExternal<RowType>;
+    label: string;
+    defaultSelected?: boolean;
+  }[];
+};
+
+export const GridFilterButtons = <RowType extends GridBaseRow>({
+  options,
+}: GridFilterButtonsProps<RowType>): JSX.Element => {
+  const [selectedOption, setSelectedOption] = useState(options.find((option) => option.defaultSelected));
```
Comment (jfee-linz) [inline src/components/gridFilter/GridFilterButtons.tsx:20]: should this default to the first item if no `defaultSelected` is configured?

File: src/components/gridFilter/GridFilterButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { useMemo, useState } from "react";
+
+import { LuiButton, LuiButtonGroup } from "@linzjs/lui";
+import { GridFilterExternal } from "../../contexts/GridContext";
+import { useGridFilter } from "../GridFilter";
+import { GridBaseRow } from "../Grid";
+
+export type GridFilterButtonsProps<RowType extends GridBaseRow> = {
+  options: {
+    filter?: GridFilterExternal<RowType>;
```
Comment (jfee-linz) [inline src/components/gridFilter/GridFilterButtons.tsx:11]: I'm a fan of alphabetizing properties

File: src/contexts/GridContextProvider.tsx [hunk 5]
```diff
@@ -24,7 +24,7 @@ export const GridContextProvider = (props: GridContextProps): ReactElement => {
   const idsBeforeUpdate = useRef<number[]>([]);
   const prePopupFocusedCell = useRef<CellPosition>();
   const [externallySelectedItemsAreInSync, setExternallySelectedItemsAreInSync] = useState(false);
-  const externalFilters = useRef<GridFilterExternal[]>([]);
+  const externalFilters = useRef<GridFilterExternal<any>[]>([]);
```
Comment (jfee-linz) [inline src/contexts/GridContextProvider.tsx:27]: What about making the context provider templated `<T = unknown>` ?

**Review by jfee-linz — CHANGES_REQUESTED** [review]

File: src/components/gridFilter/GridFilterButtons.tsx (hunk 4 above)
Comment (matttdawson) [reply src/components/gridFilter/GridFilterButtons.tsx:11]: Hmm, I usually put them in order of importance.  In this case I should have put label, filter, defaultSelected. I'll think on it.

File: src/components/gridFilter/GridFilterButtons.tsx (hunk 2 above)
Comment (matttdawson) [reply src/components/gridFilter/GridFilterButtons.tsx:29]: Nope :)

File: src/components/gridFilter/GridFilterButtons.tsx (hunk 3 above)
Comment (matttdawson) [reply src/components/gridFilter/GridFilterButtons.tsx:20]: Sure thing.

File: src/contexts/GridContextProvider.tsx (hunk 5 above)
Comment (matttdawson) [reply src/contexts/GridContextProvider.tsx]: Fixed.

File: src/components/gridFilter/GridFilterButtons.tsx (hunk 4 above)
Comment (matttdawson) [reply src/components/gridFilter/GridFilterButtons.tsx]: Done

File: src/components/gridFilter/GridFilterButtons.tsx (hunk 3 above)
Comment (matttdawson) [reply src/components/gridFilter/GridFilterButtons.tsx]: Done

File: src/components/gridFilter/GridFilterButtons.tsx (hunk 2 above)
Comment (matttdawson) [reply src/components/gridFilter/GridFilterButtons.tsx]: Done

File: src/components/gridFilter/GridFilterButtons.tsx (hunk 1 above)
Comment (matttdawson) [reply src/components/gridFilter/GridFilterButtons.tsx]: I've added luiButtonProps prop

**Review by jfee-linz — APPROVED** [review]
Looks good

**Pull request merged**